Sources/AudioKit/Audio Files/WaveformDataRequest.swift

```diff
@@ -42,17 +42,18 @@ public class WaveformDataRequest {
 
     /// will be returned on the queue you pass in or the global queue
     public func getDataAsync(with samplesPerPixel: Int,
+                             offset: Int = 0, length: Int = 0,
                              queue: DispatchQueue = DispatchQueue.global(qos: .userInitiated),
                              completionHandler: @escaping ((FloatChannelData?) -> Void)) {
         queue.async {
-            completionHandler(self.getData(with: samplesPerPixel))
+            completionHandler(self.getData(with: samplesPerPixel, offset: offset, length: length))
         }
     }
 
     /// Get waveform data
     /// - Parameter samplesPerPixel: Number of samples you want per point
```

Comment:
You have added parameters without updating the documentation comments. This is especially important since 0 and negative numbers do special things that are not obvious at all from the API. (The things that you mention in the pull request should work well for the documentation.)

Reply:
ok. I've fix parameters like as following for more clear behavior:
 * `offset`: `0` or not given (default) / `offset`: nil or minus - (from `currentFrame`) => optional `Int`
 * `length`: nil (default) or not given, for full length(or remains) / `length` >= 0 for partial retrieving. => optional`UInt`